packages/compass-home/src/components/home.tsx

```diff
@@ -411,46 +406,32 @@ function ThemedHome(
     };
   }, [appRegistry]);
 
-  const [isWelcomeOpen, setIsWelcomeOpen] = useState(showWelcomeModal);
-  const [isSettingsOpen, setIsSettingsOpen] = useState(false);
+  const [isWelcomeOpen, setIsWelcomeOpen] = useState(false);
 
-  function showSettingsModal() {
-    async function show() {
-      await preferences.ensureDefaultConfigurableUserPreferences();
-      setIsSettingsOpen(true);
+  useLayoutEffect(() => {
+    // If we haven't showed welcome modal that points users to network opt in
+    // yet, show the modal and update preferences with default values to reflect
+    // that
+    if (preferences.getPreferences().showedNetworkOptIn === false) {
+      setIsWelcomeOpen(true);
+      void preferences.ensureDefaultConfigurableUserPreferences();
```

Comment:
IIUC there is no need to do this separately when showing settings and closing the welcome (previous implementation), the only place we care about this value is on the initial component mount to either trigger the welcome modal right away and update the settings, or do nothing if opt in was shown before. This also means that potentially there is a good opportunity to remove dependency on the preferences from here completely, but as preferences are still used here to read the dark mode feature flag directly, this would've been more work, so I'm not doing this right away

Reply:
+1 This file has gotten a bit unwieldily over time, refactoring it a bit and simplifying or removing things will be nice.

Reply:
We were planning to remove this feature flag soon, right? Maybe then would be a good time to move this logic out of here too